.github/workflows/pull_request.yml

```diff
@@ -5,6 +5,16 @@ on:
       - main
 
 jobs:
+  check_python_linting:
+    name: Ruff Linting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: chartboost/ruff-action@v1
+        with:
```

Comment:
Could you confirm if the current GitHub workflow check for formatting issues? I have a vague memory that our github test only checks for linting issues.

This used to be a bug of ruff-action package, and it seems they finally fixed it here (https://github.com/ChartBoost/ruff-action?tab=readme-ov-file#use-ruff-format).

Reply:
Okay yeah, it looks like we have to add this argument such that formatting is checked. But it seems like as soon I add the argument, linting is not checked any longer, at least the output logs suggest that. So I just added it with and without the argument :thinking: